pandas/tests/copy_view/test_methods.py

```diff
@@ -214,3 +214,20 @@ def test_chained_methods(request, method, idx, using_copy_on_write):
     df.iloc[0, 0] = 0
     if not df2_is_view:
         tm.assert_frame_equal(df2.iloc[:, idx:], df_orig)
+
+
+def test_set_index(using_copy_on_write):
+    # GH 49473
+    df = DataFrame(
+        {
+            "month": [1, 4, 7, 10],
+            "year": [2012, 2014, 2013, 2014],
+            "sale": [55, 40, 84, 31],
+        }
+    )
+    df2 = df.set_index("month")
+
+    if using_copy_on_write:
+        assert np.shares_memory(get_array(df2, "year"), get_array(df, "year"))
+    else:
+        assert not np.shares_memory(get_array(df2, "year"), get_array(df, "year"))
```

Comment:
Testing that we actually did not copy in case of CoW is good, but can you also add a check that mutation does trigger CoW? So the case of "mutating df2 doesn't change df". You can do something similar as above: 

https://github.com/pandas-dev/pandas/blob/5a36b5f5273f1c19fd0c9b6ca13d985a25316649/pandas/tests/copy_view/test_methods.py#L69-L74

(but maybe leave out the additional "shares_memory" checks as in the shown snippet, that probably doesn't need to be repeated in every test)